src/EventSourcing/Domain/Policy.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EventSourcingWorkshop\EventSourcing\Domain;
+
+use EventSourcingWorkshop\Commanding\Domain\Command;
+
+/**
+ * @psalm-template HandledDomainEvent of DomainEvent
+ * @psalm-immutable
+ *
+ * Policies are meant to work as a 'given this then that' to trigger changes based on given events.
```

Comment:
An example of "given this then that" is needed.

Also to be noted that this kind of policy is limited being fired on specific events, whereas FRP (functional reactive patterns) allow for much more detailed "when these 5 things happened in precise sequence, do that"

Reply:
Provided in the example domain.